src/Hosting/Hosting/src/Internal/HostedServiceExecutor.cs

```diff
@@ -21,31 +21,17 @@ public HostedServiceExecutor(ILogger<HostedServiceExecutor> logger, IEnumerable<
             _services = services;
         }
 
-        public async Task StartAsync(CancellationToken token)
+        public Task StartAsync(CancellationToken token)
         {
-            try
-            {
-                await ExecuteAsync(service => service.StartAsync(token));
-            }
-            catch (Exception ex)
-            {
-                _logger.ApplicationError(LoggerEventIds.HostedServiceStartException, "An error occurred starting the application", ex);
-            }
+            return ExecuteAsync(service => service.StartAsync(token));
         }
 
-        public async Task StopAsync(CancellationToken token)
+        public Task StopAsync(CancellationToken token)
         {
-            try
-            {
-                await ExecuteAsync(service => service.StopAsync(token));
-            }
-            catch (Exception ex)
-            {
-                _logger.ApplicationError(LoggerEventIds.HostedServiceStopException, "An error occurred stopping the application", ex);
-            }
+            return ExecuteAsync(service => service.StopAsync(token), throwOnFirstFailure: false);
         }
 
-        private async Task ExecuteAsync(Func<IHostedService, Task> callback)
+        private async Task ExecuteAsync(Func<IHostedService, Task> callback, bool throwOnFirstFailure = true)
```

Comment:
What happened to overloads instead of default parameters?

Reply:
This thing is private, why complicate it with overloads?